app/models/concerns/canonical_email.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module CanonicalEmail
+  extend ActiveSupport::Concern
+
+  included do
+    normalizes :email, with: ->(value) { canonicalize(value) }
+  end
+
+  class_methods do
+    def canonicalize(email)
+      Mail::Address
+        .new(email)
+        .then { |address| [canonical_username(address.local), address.domain] }
+        .join('@')
+        .downcase
+    rescue Mail::Field::FieldError
+      email
```

Comment:
Not sure when this would be raised, but this seems like a very different behavior from the original implementation. I realize that one did not have any error handling at all and I suspect we are rather confident that we will always have something email-shaped here, but I am not sure why :thinking: 

Did you compare old and new behavior with invalid inputs? Would this be worth adding a spec or two?

Reply:
... I could also remove the `Mail::Address` addition here, and contemplate adding back in as part of either of those validation condition refactors, which would reduce changes here.